src/fastify-plugins/maps/offline-fallback-map.js

```diff
@@ -0,0 +1,117 @@
+import path from 'path'
+import fs from 'fs/promises'
+import FastifyStatic from '@fastify/static'
+import fp from 'fastify-plugin'
+
+import {
+  NotFoundError,
+  createStyleJsonResponseHeaders,
+  getFastifyServerAddress,
+} from '../utils.js'
+
+export const PLUGIN_NAME = 'mapeo-static-maps'
+
+export const plugin = fp(offlineFallbackMapPlugin, {
+  fastify: '4.x',
+  name: PLUGIN_NAME,
+})
+
+/**
+ * @typedef {object} OfflineFallbackMapPluginOpts
+ * @property {string} [prefix]
+ * @property {string} styleJsonPath
+ * @property {string} sourcesDir
+ */
+
+/**
+ * @typedef {object} FallbackMapPluginDecorator
+ * @property {(serverAddress: string) => Promise<any>} getResolvedStyleJson
+ * @property {() => Promise<import('node:fs').Stats>} getStyleJsonStats
+ */
+
+/** @type {import('fastify').FastifyPluginAsync<OfflineFallbackMapPluginOpts>} */
+async function offlineFallbackMapPlugin(fastify, opts) {
+  const { styleJsonPath, sourcesDir } = opts
+
+  fastify.decorate(
+    'mapeoFallbackMap',
+    /** @type {FallbackMapPluginDecorator} */
+    ({
+      async getResolvedStyleJson(serverAddress) {
+        const rawStyleJson = await fs.readFile(styleJsonPath, 'utf-8')
+        const styleJson = JSON.parse(rawStyleJson)
+
+        const sources = styleJson.sources || {}
+
+        const sourcesDirFiles = await fs.readdir(sourcesDir, {
+          withFileTypes: true,
+        })
+
+        for (const file of sourcesDirFiles) {
+          // Only work with files
+          if (file.isDirectory()) continue
```

Comment:
Nit: this is good but I think we also want to skip other file types like FIFOs and sockets. Could we get away with only keeping regular files?

```suggestion
          if (!file.isFile()) continue
```